backend/src/main/java/ca/mcgill/purposeful/controller/IdeaController.java

```diff
@@ -93,4 +97,19 @@ public IdeaDTO modifyIdea(
             iconUrlId);
     return new IdeaDTO(modifiedIdea);
   }
+
+  /**
+   * Remove an idea by its id
+   *
+   * @param id the idea's id
+   * @return a response entity with a message instance and the HttpStatus
+   */
+  @DeleteMapping({"/{id}", "/{id}/"})
+  @PreAuthorize("hasAnyAuthority('User', 'Moderator', 'Owner')")
+  public ResponseEntity<String> removeIdea(@PathVariable String id) {
+    // call service layer
+    ideaService.removeIdeaById(id);
```

Comment:
I tried to test it and if an idea has reactions, the delete will fail because of foreign key issues with reaction. 
Is it because you did not add the `@OnDelete`?

Reply:
It looks like the change I made to Reaction didn't get merged... I added it back on the branch. Can you try again?

Reply:
I had to go into psql to completely drop the database and create it again so that the database is updated to your changes.

But now it works!

Reply:
Nice :)